primitives/fee-market/src/lib.rs

```diff
@@ -42,43 +41,39 @@ impl<AccountId, Balance> Relayer<AccountId, Balance> {
 		}
 	}
 }
-
-impl<AccountId: Parameter, Balance: PartialOrd> PartialOrd for Relayer<AccountId, Balance> {
+impl<AccountId, Balance> PartialOrd for Relayer<AccountId, Balance>
+where
+	AccountId: PartialEq,
+	Balance: Ord,
+{
 	fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
-		if self.fee == other.fee {
-			return other.collateral.partial_cmp(&self.collateral);
-		}
-		self.fee.partial_cmp(&other.fee)
+		Some(match self.fee.cmp(&other.fee) {
+			// We reverse the order here to turn the collateral value into rank.
+			//
+			// Use `other.cmp(self)` instead of `self.cmp(other)`.
+			Ordering::Equal => other.collateral.cmp(&self.collateral),
```

Comment:
Why not `partial_cmp` here?  Actually, We do not need a match here, `==` equal to `Ordering::Equal`, just like the original style.

Reply:
1. Use `cmp` is more logical
2. Less calculating works, the origin code call `cmp` twice